miniapps/navier/navier_kovasznay.cpp

```diff
@@ -0,0 +1,212 @@
+   double t_final = 10 * 0.001;
+   double dt = 0.001;
+   double reference_pressure = 0.0;
+   double reynolds = 1.0 / kinvis;
+   double lam = 0.5 * reynolds
+                - sqrt(0.25 * reynolds * reynolds + 4.0 * M_PI * M_PI);
+   bool pa = true;
+   bool ni = false;
+   bool visualization = false;
+   bool checkres = false;
+} ctx;
+
+void vel_kovasznay(const Vector &x, double t, Vector &u)
+{
+   double xi = x(0);
+   double yi = x(1);
+
+   u(0) = 1.0 - exp(ctx.lam * xi) * cos(2.0 * M_PI * yi);
+   u(1) = ctx.lam / (2.0 * M_PI) * exp(ctx.lam * xi) * sin(2.0 * M_PI * yi);
+}
+
+double pres_kovasznay(const Vector &x, double t)
+{
+   double xi = x(0);
+
+   return 0.5 * (1.0 - exp(2.0 * ctx.lam * xi)) + ctx.reference_pressure;
+}
+
+int main(int argc, char *argv[])
+{
+   MPI_Session mpi(argc, argv);
+
+   OptionsParser args(argc, argv);
+   args.AddOption(&ctx.ser_ref_levels,
+                  "-rs",
+                  "--refine-serial",
+                  "Number of times to refine the mesh uniformly in serial.");
+   args.AddOption(&ctx.order,
+                  "-o",
+                  "--order",
+                  "Order (degree) of the finite elements.");
+   args.AddOption(&ctx.dt, "-dt", "--time-step", "Time step.");
+   args.AddOption(&ctx.t_final, "-tf", "--final-time", "Final time.");
+   args.AddOption(&ctx.pa,
+                  "-pa",
+                  "--enable-pa",
+                  "-no-pi",
+                  "--disable-pi",
+                  "Enable partial assembly.");
+   args.AddOption(&ctx.ni,
+                  "-ni",
+                  "--enable-ni",
+                  "-no-ni",
+                  "--disable-ni",
+                  "Enable numerical integration rules.");
+   args.AddOption(&ctx.visualization,
+                  "-vis",
+                  "--visualization",
+                  "-no-vis",
+                  "--no-visualization",
+                  "Enable or disable GLVis visualization.");
+   args.AddOption(
+      &ctx.checkres,
+      "-cr",
+      "--checkresult",
+      "-no-cr",
+      "--no-checkresult",
+      "Enable or disable checking of the result. Returns -1 on failure.");
+   args.Parse();
+   if (!args.Good())
+   {
+      if (mpi.Root())
+      {
+         args.PrintUsage(mfem::out);
+      }
+      MPI_Finalize();
+      return 1;
+   }
+   if (mpi.Root())
+   {
+      args.PrintOptions(mfem::out);
+   }
+
+   Mesh *mesh = new Mesh(2, 4, Element::QUADRILATERAL, false, 1.5, 2.0);
+
+   mesh->EnsureNodes();
+   GridFunction *nodes = mesh->GetNodes();
+   *nodes -= 0.5;
+
+   for (int i = 0; i < ctx.ser_ref_levels; ++i)
+   {
+      mesh->UniformRefinement();
+   }
+
+   if (mpi.Root())
+   {
+      std::cout << "Number of elements: " << mesh->GetNE() << std::endl;
+   }
+
+   auto *pmesh = new ParMesh(MPI_COMM_WORLD, *mesh);
+   delete mesh;
+
+   // Create the flow solver.
+   NavierSolver flowsolver(pmesh, ctx.order, ctx.kinvis);
+   flowsolver.EnablePA(true);
```

Comment:
PA should be enabled only when ctx.pa is true.
Also, the NI option is missing: flowsolver.EnableNI(ctx.ni);